compiler/rustc_middle/src/ty/context.rs

```diff
@@ -1392,6 +1392,8 @@ pub struct GlobalCtxt<'tcx> {
     /// Stores memory for globals (statics/consts).
     pub(crate) alloc_map: interpret::AllocMap<'tcx>,
 
+    pub stack_protector: Lock<FxHashSet<DefId>>,
```

Comment:
This is not correct for incremental (it's good enough to collect some data but would have to be fixed before merging). One implementation strategy I would try is to use the same mechanism as the (I forgot how exactly it's called) inferred function attrs thing pcwalton implemented a while ago.

Reply:
Yea, I think so. I'm trying to think of a more convenient way than adding a new query

Reply:
Thanks! I will learn the strategy you mentioned.